ebu_tt_live/bindings/converters/ebutt3_ebuttd.py

```diff
@@ -24,11 +24,29 @@
 DEFAULT_CELL_FONT_SIZE = CellFontSizeType('1c')
 
 
+def roundAndStrip(value, dp):
+    """
+    Round the supplied value to dp decimal places and tidy the end.
+
+    After rounding, strips any trailing zeros after the decimal point,
+    and the decimal point itself if that's the last thing we're left with.
+    """
+    if (dp < 0):
+        value = round(value, dp)
+        dp = 0
+    template = '{{:.{}f}}'.format(dp)
+    rv = template.format(value)
```

Comment:
This formatting seems a little complex. Why can't we also use `round` for values of `dp` above 0?

Reply:
That would be additional code - you still need to do the string format conversion to be able to strip the trailing zeros, and you need the decimal point there to do that safely. This way we get both behaviours in one line. OK 2 lines. I could have made it:

```python
    rv = '{{:.{}f}}'.format(dp).format(value)
```

but I thought that would be less readable.

Reply:
I was going to suggest something like
```python
rv = str(round(value, dp))
if (dp > 0):
    rv = rv.rstrip('0').rstrip('.')
return rv
```
But that doesn't work because it strips out the significant zeroes if you give it an integer `value` and a `dp` > 0.

This is one of those things that makes me feel like there should be a simple solution, but your implementation seems to work fine, so I'm not going to fuss about it anymore.